src/algebra/direct_sum/internal.lean

```diff
@@ -41,7 +41,8 @@ internally graded ring
 
 open_locale direct_sum big_operators
 
-variables {ι : Type*} {S R : Type*} [decidable_eq ι]
+variables {ι : Type*} {S R : Type*} [deq : decidable_eq ι]
+include deq
```

Comment:
I'd be inclined to instead do
```suggestion
variables {ι : Type*} {S R : Type*}
section direct_sum
variables [decidable_eq ι]
```
and then put all the `homogenous` stuff in a new section right at the bottom of the file, rather than juggling `deq` throughout.